mergify_engine/queue/freeze.py

```diff
@@ -131,10 +132,25 @@ async def save(self) -> None:
             ),
         )
 
+        await self._refresh_pulls(source="internal/queue_freeze_create")
+
     async def delete(self) -> bool:
-        return bool(
+
+        result = bool(
             await self.repository.installation.redis_queue.hdel(
                 self._get_redis_hash(self.repository),
                 self._get_redis_key(self.repository, self.name),
             )
         )
+
+        await self._refresh_pulls(source="internal/queue_freeze_delete")
+
+        return result
+
+    async def _refresh_pulls(self, source: str) -> None:
+
+        async for train in merge_train.Train.iter_trains(self.repository):
+            await super(type(train), train).refresh_pulls(
```

Comment:
```suggestion
            await train.refresh_pulls(
```

This should be enough, no ?